lib/did_you_mean/finders/name_error_finders/similar_class_finder.rb

```diff
@@ -32,6 +33,13 @@ def scopes
 
     private
 
+    def autoload_class_name_inflector
+      return unless defined?(Rails)
+      name_is_singular = name_from_message.to_s.pluralize.singularize == name_from_message.to_s
+      class_name_inflector = name_is_singular ? name_from_message.to_s.pluralize : name_from_message.to_s.singularize
+      class_name_inflector.safe_constantize
+    end
+
```

Comment:
I don't like this change. Debugging tools should never have side effects. This side effect would confuse people who know how Rails loads constants. If you are using Rails, you should be aware of it.